tests/test_codebase/test_mmdet/test_object_detection_model.py

```diff
@@ -385,34 +385,43 @@ def partition1_postprocess(self, *args, **kwargs):
         assert_forward_results(results, 'PartitionTwoStageModel')
 
 
-data_cfg1 = mmcv.Config(
-    dict(
-        data=dict(
-            test=dict(type='CocoDataset'),
-            val=dict(type='CityscapesDataset'),
-            train=dict(type='CityscapesDataset'))))
-data_cfg2 = mmcv.Config(
-    dict(
-        data=dict(
-            val=dict(type='CocoDataset'), train=dict(
-                type='CityscapesDataset'))))
-data_cfg3 = mmcv.Config(dict(data=dict(train=dict(type='CocoDataset'))))
-data_cfg4 = mmcv.Config(dict(data=dict(error=dict(type='CocoDataset'))))
-
-
-@pytest.mark.parametrize('cfg', [data_cfg1, data_cfg2, data_cfg3, data_cfg4])
-def test_get_classes_from_cfg(cfg):
-    from mmdet.datasets import DATASETS
-
-    from mmdeploy.codebase.mmdet.deploy.object_detection_model import \
-        get_classes_from_config
-
-    if 'error' in cfg.data:
-        with pytest.raises(RuntimeError):
-            get_classes_from_config(cfg)
-    else:
-        assert get_classes_from_config(
-            cfg) == DATASETS.module_dict['CocoDataset'].CLASSES
+class TestGetClassesFromCfg:
+    data_cfg1 = mmcv.Config(
+        dict(
+            data=dict(
+                test=dict(type='CocoDataset'),
+                val=dict(type='CityscapesDataset'),
+                train=dict(type='CityscapesDataset'))))
+
+    data_cfg2 = mmcv.Config(
+        dict(
+            data=dict(
+                val=dict(type='CocoDataset'),
+                train=dict(type='CityscapesDataset'))))
+    data_cfg3 = mmcv.Config(dict(data=dict(train=dict(type='CocoDataset'))))
+    data_cfg4 = mmcv.Config(dict(data=dict(error=dict(type='CocoDataset'))))
+
+    @pytest.mark.parametrize('cfg',
+                             [data_cfg1, data_cfg2, data_cfg3, data_cfg4])
+    def test_get_classes_from_cfg(self, cfg):
+        from mmdet.datasets import DATASETS
+        from mmdeploy.codebase.mmdet.deploy.object_detection_model import \
+            get_classes_from_config
+
+        if 'error' in cfg.data:
```

Comment:
I don't understand where this error comes from？

Reply:
In `data_cfg4`. The key of inner dict is "error".